### PR TITLE
sidebarFocus and panelFocus, part of #59858

### PR DESCRIPTION
**Pull request by octref**

### CODE REVIEW

**Comment by octref** [comment]
@isidorn There is one bug that I didn't fix, but I guess it's related to View's underlying implementation.

When you run prev/next panel view and go to output view, the focus will be lost from panel area.
The same thing for running prev/next sidebar view to an un-initialized git viwelet.

Do you mind taking a look?

Working as expected:

keyboard-nav

Except when going into Ouput or an uninitialized Git Viewlet. The focus is lost. I guess because the empty View is un-focusable?

image

I do hope the focus problem can be fixed, so I can finally navigate around the workbench without touching touchpad.

**Comment by octref** [comment]
@isidorn As the Git Viewlet is empty, it would be un-focusable. @joaomoreno will fix that. Do you mind fixing the same thing for output?
Whenever I run Previous/Next panel that ends up in Output panel, focus will be lost from Panel. I'm calling `openPanel` with focus set to true: https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/browser/parts/panel/panelActions.ts#L205

File: src/vs/workbench/browser/parts/panel/panelPart.ts [hunk 1]
```diff
@@ -181,7 +198,11 @@ export class PanelPart extends CompositePart<Panel> implements IPanelService {
 			}
 		}
 
-		return promise.then(() => this.openComposite(id, focus));
+		return promise.then(async () => {
+			const composite = await this.openComposite(id, focus);
```
Comment (isidorn) [inline src/vs/workbench/browser/parts/panel/panelPart.ts]: Why is this needed? Why can't we simply change the contextKey based on focus tracker only?

File: src/vs/workbench/browser/parts/panel/panelPart.ts [hunk 2]
```diff
@@ -27,13 +27,15 @@ import { CompositeBar } from 'vs/workbench/browser/parts/compositeBar';
 import { ToggleCompositePinnedAction } from 'vs/workbench/browser/parts/compositeBarActions';
 import { IBadge } from 'vs/workbench/services/activity/common/activity';
 import { INotificationService } from 'vs/platform/notification/common/notification';
-import { Dimension } from 'vs/base/browser/dom';
+import { Dimension, trackFocus } from 'vs/base/browser/dom';
 import { localize } from 'vs/nls';
 import { IDisposable } from 'vs/base/common/lifecycle';
 import { RawContextKey, IContextKey, IContextKeyService } from 'vs/platform/contextkey/common/contextkey';
 
 const ActivePanleContextId = 'activePanel';
```
Comment (isidorn) [inline src/vs/workbench/browser/parts/panel/panelPart.ts]: I do not see the point of these consts, `ActivePanleContextId` and `PanelFocusContextId` why don't we simply inline them, we are using them only once?

**Comment by isidorn** [comment]
@octref I have reviewed the PR, can you please tackle my comments.
As for the output focus issue I will fix it today.

**Comment by isidorn** [comment]
Thanks. Looks good, let's merge it in
